test/modules/paymentProcessor/bridging/abstracts/PP_Connext_Crosschain_v1_Test.t.sol

```diff
@@ -0,0 +1,547 @@
+
+        _setUpOrchestrator(paymentProcessor);
+        _authorizer.setIsAuthorized(address(this), true);
+
+        // Initialize payment processor with config
+        bytes memory configData = abi.encode(mockEverClearSpoke, mockWeth);
+        paymentProcessor.init(_orchestrator, _METADATA, configData);
+
+        // Deploy and add payment client through timelock process
+        impl = address(new ERC20PaymentClientBaseV1Mock());
+        paymentClient = ERC20PaymentClientBaseV1Mock(Clones.clone(impl));
+        _orchestrator.initiateAddModuleWithTimelock(address(paymentClient));
+        vm.warp(block.timestamp + _orchestrator.MODULE_UPDATE_TIMELOCK());
+        _orchestrator.executeAddModule(address(paymentClient));
+
+        // Configure payment client
+        paymentClient.init(_orchestrator, _METADATA, bytes(""));
+        paymentClient.setIsAuthorized(address(paymentProcessor), true);
+        paymentClient.setToken(token);
+
+        _setupInitialBalances();
+    }
+
+    /* Test initialization
+        └── When the contract is initialized
+            ├── Then it should set the orchestrator correctly
+            └── Then it should set up Connext configuration properly
+    */
+    function testInit() public override(ModuleTest) {
+        assertEq(
+            address(paymentProcessor.orchestrator()), address(_orchestrator)
+        );
+    }
+
+    /* Test interface support
+        ├── When checking IModule_v1 interface
+        │   └── Then it should return true
+        ├── When checking ICrossChainBase_v1 interface
+        │   └── Then it should return true
+        └── When checking random interface
+            └── Then it should return false
+    */
+    function testSupportsInterface() public {
+        // Test for IModule_v1 interface
+        assertTrue(
+            paymentProcessor.supportsInterface(type(IModule_v1).interfaceId)
+        );
+
+        // Test for ICrossChainBase_v1 interface
+        assertTrue(
+            paymentProcessor.supportsInterface(
+                type(ICrossChainBase_v1).interfaceId
+            )
+        );
+
+        // Test for a non-supported interface (using a random interface ID)
+        assertFalse(paymentProcessor.supportsInterface(0xffffffff));
+    }
+
+    /* Test reinitialization protection
+        └── When trying to reinitialize an already initialized contract
+            └── Then it should revert with InvalidInitialization
+    */
+    function testReinitFails() public override(ModuleTest) {
+        vm.expectRevert(OZErrors.Initializable__InvalidInitialization);
+        paymentProcessor.init(_orchestrator, _METADATA, abi.encode(1));
+    }
+
+    /* Test single payment processing
+        ├── Given a valid payment order
+        │   ├── When processing through Connext bridge
+        │   │   ├── Then it should emit PaymentProcessed event
+        │   │   ├── Then it should transfer tokens correctly
+        │   │   └── Then it should create Connext intent
```

Comment:
Good point, in our mock we're changing the status of the intent to ADDED but we should be verifying that in the test.

```

        // Set intent status to ADDED and emit the event
        status[_intentId] = IntentStatus.ADDED;
        
        ```

Reply:
I think processPayments will need to return the intentID, at least for the tests, this will allow us to check if the status has been changed in the mock

Reply:
@leeftk this is a general comment for testing the intentID

Reply:
I added this as we discussed, there's now a mapping in the payment process that stores the payment -> receive -> bridge data. We're now getting that value from the payment processor, and then checking that the status in the mock contract for that intentID has flipped to added. Please take a look at this to see if it follow what we discuss on the call, any feedback is welcome.